spring-cloud-aws-sns/src/test/java/io/awspring/cloud/sns/integration/SnsTemplateIntegrationTest.java

```diff
@@ -106,4 +116,26 @@ void send_validPersonObject_usesTopicChannel_send_arn_read_sqs() {
 		});
 	}
 
+	@Test
+	void send_test_message_for_existing_topic_name_trigger_trigger_iteration() {
+		ListTopicArnResolver listTopicArnResolver = new ListTopicArnResolver(snsClient);
+		SnsTemplate snsTemplateTestCache = new SnsTemplate(snsClient, listTopicArnResolver, null);
+		snsTemplateTestCache.sendNotification(TOPIC_NAME + 110, "message content", "subject");
+	}
+
+	@Test
+	void send_test_message_for_not_existing_topic_name() {
+		ListTopicArnResolver listTopicArnResolver = new ListTopicArnResolver(snsClient);
+		SnsTemplate snsTemplateTestCache = new SnsTemplate(snsClient, listTopicArnResolver, null);
+		assertThatThrownBy(
+				() -> snsTemplateTestCache.sendNotification("Some_random_topic", "message content", "subject"))
+						.isInstanceOf(NotExistentTopicException.class);
+	}
+
+	private static void createTopics() {
+		for (int i = 0; i < 300; i++) {
```

Comment:
Lets keep this number as low as possible - higher than the single response page size, but not so large as it only extends time needed to run tests.